src/KDSoapClient/KDSoapAuthentication.cpp

```diff
@@ -91,3 +133,57 @@ void KDSoapAuthentication::handleAuthenticationRequired(QNetworkReply *reply, QA
         reply->setProperty("authAdded", true);
     }
 }
+
+bool KDSoapAuthentication::hasWSUsernameTokenHeader() const
+{
+    return hasAuth() && d->useWSUsernameToken;
+}
+
+void KDSoapAuthentication::writeWSUsernameTokenHeader(KDSoapNamespacePrefixes &namespacePrefixes, QXmlStreamWriter &writer, const QString &messageNamespace, bool forceQualified) const
+{
+    Q_UNUSED(namespacePrefixes);
```

Comment:
Why have those arguments then, if they are unused? Can't they be removed?

Reply:
I used the same function signature as KDSoapMessageAddressingProperties::writeMessageAddressingProperties, however that is not actually necessary.